### PR TITLE
Use serviceerror.Unavailable error type instead of serviceerror.Internal

### PR DESCRIPTION
**Pull request by wxing1292**
**What changed?**
* Use serviceerror.Unavailable error type instead of serviceerror.Internal
  * serviceerror.Unavailable indicates errors like service's dependency is down, retryable
  * serviceerror.Internal indicates errors like NPE in service logic, non-retryable

**Why?**
Let SDK do better error handling

**How did you test it?**
Existing tests

**Potential risks**
SDKs (go & java) currently does not differentiate the different between Unavailable & Internal, both are retried.
SDKs in the future should only retry on Unavailable, and try not to retry on Internal

**Is hotfix candidate?**
N/A

### CODE REVIEW

**Comment by wxing1292** [comment]
This PR is part 0, there will be following PRs which return internal error if internal service contract is broken
* expect some flag is set but actually not set
* NPE

**Review by alexshtin — APPROVED** [review]

File: common/archiver/filestore/historyArchiver.go [hunk 1]
```diff
@@ -206,7 +206,7 @@ func (h *historyArchiver) Get(
 	dirPath := URI.Path()
 	exists, err := directoryExists(dirPath)
 	if err != nil {
-		return nil, serviceerror.NewInternal(err.Error())
+		return nil, serviceerror.NewUnavailable(err.Error())
```
Comment (Sushisource) [inline common/archiver/filestore/historyArchiver.go:209]: This seems like an internal error still?

File: common/archiver/filestore/historyArchiver.go [hunk 2]
```diff
@@ -226,7 +226,7 @@ func (h *historyArchiver) Get(
 	} else {
 		highestVersion, err := getHighestVersion(dirPath, request)
 		if err != nil {
-			return nil, serviceerror.NewInternal(err.Error())
+			return nil, serviceerror.NewUnavailable(err.Error())
```
Comment (Sushisource) [inline common/archiver/filestore/historyArchiver.go:229]: This one too? Does `getHighestVersion` reach out to the network?

**Comment by Sushisource** [comment]
So I think this is OK as long as long as the follow up (ideally before another release) is to go make the things that should be internal interal, of which there are definitely a lot.

Ex: Just a few files I found that look to be entirely internal rather than unavailable errors:
* https://github.com/temporalio/temporal/pull/1916/files#diff-5539238bc6588be646a39da55e4ad808a5400aa28162767bc4556321766ff262
* https://github.com/temporalio/temporal/pull/1916/files#diff-a25b2e4dcfa5108412bbd426634c31fd936028bc86d719507f5c4c74a5e7ac99
* https://github.com/temporalio/temporal/pull/1916/files#diff-8037283c80ca983dab71e2fe828e62beab68832d68b16ac51488af45d0955f06
* https://github.com/temporalio/temporal/pull/1916/files#diff-8788caed195a799e11b26c907ea0f1723337b55e6a77d55817b14755e7c4cb4f

etc

It might make sense to add the changes back to internal as chunked commits to this PR for readability.

**Pull request merged**